src/components/select/SelectItemCommons.tsx

```diff
@@ -42,13 +42,30 @@ function withSelectItemWrapper<P extends SelectOption>(
   optionType: SelectOptionType,
 ) {
   const Wrapper = forwardRef<HTMLDivElement, P>((props, ref) => {
-    const { helpText, label, prefix, suffix, value, ...rest } = props;
+    const { helpText, label, prefix, selected, suffix, value, ...mantineAndCustomProps } = props;
```

Comment:
I've explicitly extracted `selected` because it's the only one that `SelectItemStatus` needs

Reply:
I think I would only extract our own implementation stuff here: helpText, prefix, suffix

so that all the option fields end up in `mantineAndCustomProps` and you don't have to add them back after in both components